Improve workflow in docs/google-kubernetes-tutorial.md (#400)
pingcap/tidb-operator: charts/tidb-cluster/templates/NOTES.txt

    kubectl port-forward -n {{ .Release.Namespace }} svc/{{ template "cluster.name" . }}-tidb 4000:4000 &
{{- if .Values.tidb.passwordSecretName }}
    mysql -h 127.0.0.1 -P 4000 -u root -D test -p
{{- else }}
    mysql -h 127.0.0.1 -P 4000 -u root -D test
  Set a password for your user
    SET PASSWORD FOR 'root'@'%' = '{{ (randAlphaNum 10) }}'; FLUSH PRIVILEGES;
{{- end -}}
{{- if .Values.monitor.create }}
* View monitor dashboard for TiDB cluster
   kubectl port-forward -n {{ .Release.Namespace }} svc/{{ template "cluster.name" . }}-grafana 3000:3000
   Open browser at http://localhost:3000. The default username and password is admin/admin.

>    If you are running this from a remote machine, you must specify server's external IP address.

Reviewer: ```suggestion
   If you are running this from a remote machine, you must specify the server's external IP address.
```

    kubectl port-forward -n {{ .Release.Namespace }} svc/{{ template "cluster.name" . }}-tidb 4000:4000 &
{{- if .Values.tidb.passwordSecretName }}
    mysql -h 127.0.0.1 -P 4000 -u root -D test -p
{{- else }}
    mysql -h 127.0.0.1 -P 4000 -u root -D test
  Set a password for your user
    SET PASSWORD FOR 'root'@'%' = '{{ (randAlphaNum 10) }}'; FLUSH PRIVILEGES;
{{- end -}}
{{- if .Values.monitor.create }}
* View monitor dashboard for TiDB cluster
   kubectl port-forward -n {{ .Release.Namespace }} svc/{{ template "cluster.name" . }}-grafana 3000:3000
   Open browser at http://localhost:3000. The default username and password is admin/admin.

>    Note that, if you run in remote server, you need to specify server's external IP address.

Reviewer 2: ```suggestion
   If you are running this from a remote machine, you must specify server's external IP address.
```

>    In gcloud web console, you can user web preview feature, e.g. https://ssh.cloud.google.com/devshell/proxy?port=3000

Reviewer 3: For GCP, I think it makes sense to use a public load balancer for Grafana service. And access the dashboard using the load balancer endpoint.

Reviewer 2: A public load balancer would be insecure.

Reviewer 2: We should only show this note for GCP users.

Author: There are two ways to access the grafana dashboard:

- local `kubectl port-forward`
- cluster service or ingress, etc

These notes are added for the first scenario in case users are not running this from local machine.

Note about `gcloud web console` is too specific, it seems not necessary. What do you think?

For the second scenario, we can support it later in charts. Show grafana access URL if you specific external IP address or ingress domain. However, it is not possible if the external IP address or domain are allocated by k8s.

Reviewer 3: Ah, we haven't instructions about how to view the dashboard in GKE tutorial. I think it's better documented there.

Author: > We should only show this note for GCP users.

Not sure how to detect this in helm charts.

> Ah, we haven't instructions about how to view the dashboard in GKE tutorial. I think it's better documented there.

You mean to add a section about how to view the grafana dashboard in `google-kubernetes-tutorial.md`?

Reviewer 3: I think we don't have to specify these instructions in the `NOTES.txt`. Instead, we should add platform-specific instructions in their own documentation.

Author: deleted
